File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    data.sort(key = lambda x : (x[col-1], -x[0]))
    nums = []
    n = len(data[0])
    for i in range(row_begin, row_end+1):
        tmp = 0
        for j in range(n):
            tmp += data[i-1][j] % i
        nums.append(tmp)
    answer = nums[0]
    for i in range(1, len(nums)):
        answer = answer ^ nums[i]
    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.Arrays;
import java.util.ArrayList;
class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
        Arrays.sort(data, (o1, o2) -> (o1[col-1] == o2[col-1] ? o2[0]-o1[0] : o1[col-1] - o2[col-1]));
        int n = data[0].length;
        ArrayList<Integer> nums = new ArrayList<>();
        for (int i = row_begin; i <= row_end; i++) {
            int tmp = 0;
            for (int j = 0; j < n; j++) {
                tmp += data[i-1][j] % i;
            }
            nums.add(tmp);
        }
        answer = nums.get(0);
        
        for (int i = 1; i < nums.size(); i++) {
            answer = answer ^ nums.get(i);
        }
        
        
        return answer;
    }
}